client/src/components/DeletePledgeConfirmation/index.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import Footer from '../Footer'
+import BackButton from '../BackButton'
+import axios from 'axios'
+import './style.css'
+
+const DeletePledgeConfirmation = (props) => {
+
+
+
+
+  const { pledgeId } = props.match.params
+  //get pledge info and handleChange function of the pledge to change the pledgeExit state in the pledgePage
+  let { pledge_name, pledge_image } = props.location
+  let { history } = props
+
+  //  user_id will be changed to be taken from authentication cookies
+  //  handleChange function is called to change the state of pledge
+  const deleteUserPledge = () => {
+    const userId = 1;
+    axios.get(`/api/${userId}/${pledgeId}/deletePledge`).then(
+      () => {
+        history.push(`/home/${userId}`)
+      }
+    ).catch(err => { console.log("axios err", err) })
+  }
+
+
+  if (!pledge_name) {
+    const data = sessionStorage.getItem('pledgeInfo')
+    pledge_name = JSON.parse(data).pledge_name
+    pledge_image = JSON.parse(data).pledge_image
+  }
```

Comment:
You can use the destructrue of already defined variable as followes :
`( {peldge_name,pledge_image}  = json.parse(data))`
Soruce : https://stackoverflow.com/questions/32138513/how-to-destructure-an-object-to-an-already-defined-variable